### ISSUE TITLE
Can this be used with multiple threads?

### ISSUE THREAD

**Issue by jjpepper** [invenia/std-semaphore#1]
Can this class be used with multiple threads?  If so, can you give an example of how to do that?  If not, isn't it of pretty limited use?

**Comment by lefth** [comment]
@jjpepper Yes. Wrap it in an `Arc` (`let semaphore = Arc::new(Semaphore::new(8))`). Before the `Arc` is moved, it must be cloned (outside the scope of a move closure or async move block): `let semaphore = Arc::clone(&semaphore)`.
